p2p/src/message_types.rs

```diff
@@ -1,36 +1,48 @@
+use lazy_static::lazy_static;
 use std::collections::HashMap;
+lazy_static! {
+    static ref MSG_TYPES: HashMap<u16, &'static str> = get_message_types();
+}
 
-pub fn get_message_types() -> HashMap<u16, &'static str> {
+fn get_message_types() -> HashMap<u16, &'static str> {
     let mut message_types = HashMap::new();
 
     message_types.insert(0, "Raw / Invalid");
+    message_types.insert(0x22, "Sync Request");
+    message_types.insert(0x23, "Sync Close");
     message_types.insert(0x01, "Sync Acknowledged");
     message_types.insert(0x05, "Send Block");
-    message_types.insert(0x0a, "Get Block");
-    message_types.insert(0x1a, "Handshake");
-    message_types.insert(0x1b, "Send Chain Digest");
-    message_types.insert(0x1c, "Send Chain Digest");
-    message_types.insert(0x22, "Sync Request");
-    message_types.insert(0x45, "Send Block Count");
-    message_types.insert(0x46, "Got Block Count");
-    message_types.insert(0x60, "Get Chain List");
-    message_types.insert(0x61, "Got Chain List");
+    message_types.insert(0x0a, "Block");
+    message_types.insert(0x0b, "Block Acknowledged");
+    message_types.insert(0x0c, "Block Reject"); // message will contain a code explaining why the block was rejected
+                                                // 4 part handshake
+    message_types.insert(0x0a, "Handshake Init");
+    message_types.insert(0xa1, "Handshake Init Response");
+    message_types.insert(0xa2, "Handshake Test Key");
+    message_types.insert(0xa3, "Handshake Confirm Key Test");
+
+    // chain digests
+    message_types.insert(0x1b, "Send Chain Digest (ask)"); // - why are there two? Backward combatability? TODO: remove one
```

Comment:
I found both in the old code, you're free to remove one :)

Reply:
Going to leave for now